packages/react/src/primitives/TextField/TextField.tsx

```diff
@@ -3,55 +3,41 @@ import classNames from 'classnames';
 
 import { classNameModifier } from '../shared/utils';
 import { ComponentClassNames } from '../shared/constants';
-import { splitPrimitiveProps } from '../shared/styleUtils';
 import { FieldDescription, FieldErrorMessage } from '../Field';
 import { FieldGroup } from '../FieldGroup';
 import { Flex } from '../Flex';
 import { Input } from '../Input';
 import { Label } from '../Label';
-import { PrimitiveProps, TextFieldProps } from '../types';
-import {
-  isTextAreaField,
-  isInputField,
-  isInputRef,
-  isTextAreaRef,
-} from './utils';
-import { TextArea } from '../TextArea';
+import { Primitive } from '../types';
+import { splitPrimitiveProps } from '../shared/styleUtils';
+import { TextFieldProps } from '../types';
 import { useStableId } from '../utils/useStableId';
-import { useDeprecationWarning } from '../../hooks/useDeprecationWarning';
 
-export const DEFAULT_ROW_COUNT = 3;
-
-const TextFieldPrimitive = <Multiline extends boolean>(
-  props: PrimitiveProps<TextFieldProps<Multiline>, 'input' | 'textarea'>,
-  ref: React.ForwardedRef<HTMLTextAreaElement | HTMLInputElement>
-) => {
+const TextFieldPrimitive: Primitive<TextFieldProps, 'input'> = (props, ref) => {
   const {
     className,
     descriptiveText,
     errorMessage,
     hasError = false,
     id,
+    innerEndComponent,
+    innerStartComponent,
     label,
     labelHidden = false,
     outerEndComponent,
     outerStartComponent,
-    innerStartComponent,
-    innerEndComponent,
-    isMultiline, // remove from rest to prevent passing as DOM attribute
-    type, // remove from rest to prevent passing as DOM attribute to textarea
```

Comment:
No longer need to pick this one out since we no longer render a <textarea> element.